### ISSUE TITLE
@accessible macro does not correctly generate ZStream and ZSink accessors

### ISSUE THREAD

**Issue by dpovey** [zio/zio#4223]
The test in AccessibleSpec in the git repo has the following example:

```
 @accessible
 object Module {
   def stream(arg1: Int): ZStream[Any,Nothing, Int]  
 }
```

But the check seems to expect
```
object Check {
  def stream(arg1: Int): ZIO[Has[Module.Service], Nothing, ZStream[Any, Nothing, Int]] = Module.stream(arg1)
}
```

It should be:
```
object Check {
  def stream(arg1: Int): ZStream[Has[Module.Service], Nothing, Int] = Module.stream(arg1)
}
```

ZSink also seems to have the same issue

**Comment by dpovey** [comment]
It looks like ZSink will be harder to fix as there is no `ZSink.access`

**Comment by dpovey** [comment]
This will also break the macro part of the `zio-intellij` plugin.

**Comment by iravid** [comment]
Hey @dpovey - here's an implementation of `ZSink.accessSink` that you can use:
```scala
  def accessSink[R]: AccessSinkPartiallyApplied[R] = new AccessSinkPartiallyApplied[R]

  final class AccessSinkPartiallyApplied[R](private val dummy: Boolean = true) extends AnyVal {
    def apply[E, I, L, Z](f: R => ZSink[R, E, I, L, Z]): ZSink[R, E, I, L, Z] =
      ZSink {
        for {
          env <- ZManaged.environment[R]
          push <- f(env).push
        } yield push
      }
  }
```

**Comment by dpovey** [comment]
Thanks @iravid. Also needed `ZSink.provide` implemented too which I think I got right, but I'm a little wobbly on the details.  I have added these to my PR and updated the `@accessible` macro to also generate accessors that output just `ZSink`

**Comment by adamgfraser** [comment]
Closed by #4224.
